portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -34,11 +34,11 @@
     private final Gson gson = new Gson();
 
     public static class Login{ 
-        private final String status;
+        private final Boolean status;
         private final String url;
   
-        public Login(String status, String url) {
-            this.status =status;
+        public Login(Boolean status, String url) {
```

Comment:
Prefer `boolean` over `Boolean` unless you need the value to be nullable (or need to use it in a generic list)

In java the numeric types (`boolean`, `int`, `double`, etc.) have related classes (`Boolean`, `Integer`, `Double`). For each of those classes, think of them like this:

```java
public class Integer {
   private final int value;

   public Integer(int value) { this.value = value; }

  // More methods and such...
}
```

This is called a "boxed" int/double/bool/etc, because while a plain `int` or `double` exists on the program stack, a "boxed" value exists on the program heap, meaning that it requires going through a pointer to access it, and requires a memory allocation/produces objects that need to be garbage collected.

When you assign a `boolean` to a variable whose type is `Boolean`, java will perform "auto-boxing", which is it converts to the boxed value. For boolean auto-boxing should always re-use the same two `Boolean` values and not construct new ones all the time, but for integers, auto-boxing integers over (IIRC) 100 or 1000 results in new allocations.

Conversely, when you assign a `Boolean` to a variable whose type is `boolean`, java performs auto-unboxing; it calls `Boolean.booleanValue()` for you to extract the `boolean` inner value. This can fail, since `Boolean` is a class, it could be null, which means that assigning from `Boolean` to `boolean` has a chance to throw `NullPointerException`.

Therefore, you should *always* use the lower-cased versions of the primitive types, unless you have a very specific reason to do otherwise.

Reply:
ok fixed!